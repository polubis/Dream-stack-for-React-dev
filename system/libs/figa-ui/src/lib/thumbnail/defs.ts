import type { ReactNode } from 'react';
import type { ImageProps } from '../image';

interface ThumbnailProps extends ImageProps {
  img?: (props: Omit<ThumbnailProps, 'img'>) => ReactNode;
}

export type { ThumbnailProps };
