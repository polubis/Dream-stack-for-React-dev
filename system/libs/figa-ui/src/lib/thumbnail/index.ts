export * from './thumbnail';
export * from './defs';
