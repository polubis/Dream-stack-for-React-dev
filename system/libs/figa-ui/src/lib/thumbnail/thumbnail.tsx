import { Font } from '../font';
import { Image } from '../image';
import type { ThumbnailProps } from './defs';
import c from 'classnames';

const Thumbnail = ({ title, className, img, ...props }: ThumbnailProps) => {
  return (
    <div className={c('thumbnail', className)}>
      {img ? (
        img({
          title,
          className,
          ...props,
        })
      ) : (
        <Image {...props} />
      )}
      <div className="thumbnail-content">
        <Font variant="h2">{title}</Font>
      </div>
    </div>
  );
};

export { Thumbnail };
