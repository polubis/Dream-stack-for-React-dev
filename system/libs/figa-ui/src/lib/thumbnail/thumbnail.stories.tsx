import type { Story, Meta } from '@storybook/react';

import { Thumbnail } from './thumbnail';

export default {
  component: Thumbnail,
  title: 'Thumbnail',
} as Meta;

const Template: Story = () => {
  return (
    <Thumbnail
      alt="Thumbnail"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqMoD_dTTKX7ZPPEdDaSFDzOfKk7WNfPDq8w&usqp=CAU"
      title="Example of thumbnail"
    />
  );
};

export const Default = Template.bind({});
Default.args = {};
