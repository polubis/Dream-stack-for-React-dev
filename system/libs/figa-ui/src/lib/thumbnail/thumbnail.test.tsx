import { fireEvent, render, screen } from '@testing-library/react';
import { Thumbnail } from './thumbnail';

describe('Thumbnail can be used when: ', () => {
  const SRC =
    'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqMoD_dTTKX7ZPPEdDaSFDzOfKk7WNfPDq8w&usqp=CAU';

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = render(
      <Thumbnail
        className="my-class"
        src={SRC}
        title="My thumbnail"
        alt="My thumbnail"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('allows to pass native image properties', () => {
    const spy = jest.fn();

    render(
      <Thumbnail
        src={SRC}
        title="My thumbnail"
        alt="My thumbnail"
        onClick={spy}
      />
    );

    fireEvent.click(screen.getByAltText(/My thumbnail/));

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('allows to use own image', () => {
    render(
      <Thumbnail
        src={SRC}
        title="My thumbnail"
        alt="My thumbnail"
        img={(props) => <img alt={props.alt} src={props.src} />}
      />
    );

    screen.getByAltText(/My thumbnail/);
  });
});
