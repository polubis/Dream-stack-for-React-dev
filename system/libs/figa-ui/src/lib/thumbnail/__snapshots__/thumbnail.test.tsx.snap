// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="thumbnail my-class"
  >
    <picture
      class="image"
    >
      <img
        alt="My thumbnail"
        loading="eager"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqMoD_dTTKX7ZPPEdDaSFDzOfKk7WNfPDq8w&usqp=CAU"
      />
    </picture>
    <div
      class="thumbnail-content"
    >
      <h2
        class="font h2 default"
      >
        My thumbnail
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
